lua/mini/misc.lua

```diff
@@ -409,6 +409,130 @@ MiniMisc.zoom = function(buf_id, config)
   end
 end
 
+--- Restore the cursor on file reopen
+---
+--- What it does:
+--- - When reopening a file, this will make sure the cursor is placed back to
+---   the position where you left before.
+--- - This implements :help restore-cursor in a nicer way.
+--- - Creates autocommand for |BufRead| and |FileType| event with
+---   |MiniMisc.restore_cursor|
+---
+--- If you want to lazy load this use the |BufReadPre| event.
+---
+---@param opts table Allows to set options for |MiniMisc.restore_cursor|
+---   You can set the two options, `ignore_buftype` and `ignore_filetype` as
+---   an array of strings.
+---   Default:
+---   ```
+---   {
+---     ignore_buftype = { "quickfix", "nofile", "help" },
+---     ignore_filetype = { "gitcommit", "gitrebase" },
+---     always_center = true,
+---   }
+---   ```
+---
+---@usage >
+---   require('mini.misc').setup_restore_cursor()
+MiniMisc.setup_restore_cursor = function(opts)
+  opts = opts or {}
+  opts.ignore_buftype = opts.ignore_buftype or { "quickfix", "nofile", "help" }
+  opts.ignore_filetype = opts.ignore_filetype or { "gitcommit", "gitrebase" }
+  opts.always_center = opts.always_center or true
```

Comment:
Center always with `g'"zvzz`, not behind an option.

Reply:
This is the default. We will always center. However I do not want that and also maybe other who used to the behavior of the vim plugin. So you can disable it with that option now.

Reply:
I am fine with `center` option which decided whether to do a simple `zz` or not.

All current if-else-s with their different behavior add too much complexity for little gain. They won't be a part of 'mini.misc'.

Reply:
Fixed.